modules/pubDirectSystem.js

```diff
@@ -0,0 +1,190 @@
+  },
+  /**
+   * performs action to obtain id and return a value in the callback's response argument
+   * @function
+   * @param {SubmoduleParams} [configParams]
+   * @param {ConsentData} [consentData]
+   * @param {(Object|undefined)} cacheIdObj
+   * @returns {IdResponse|undefined}
+   */
+  getId(configParams, consentData, cacheIdObj) {
+    if (!configParams || !configParams.publisherId) {
+      utils.logError(`Pubdirect params missing.`);
+      return undefined;
+    }
+    const initSync = new Promise(function(resolve, reject) {
+      // pubdirect ID logic
+      var siteIdValue = {
+        ID: '',
+        timestamp: null,
+        origin: '',
+        DoNotTrack: false,
+        consentData: consentData,
+        cacheIdObj: cacheIdObj,
+      };
+      var iframeCreated = false;
+      var incomingUserId = '';
+      var receiver = {};
+      var consent = true;
+      var doNotTrackUser = false;
+
+      window.addEventListener('message', msgFiltering);
+
+      var userRandomId = {
+        ID: IDGenerator(),
+        timestamp: new Date().getTime(),
+        origin: '',
+        DoNotTrack: false,
+      };
+
+      if (iframeCreated === false) {
+        var iframe = document.createElement('iframe');
+        iframe.style.display = 'none';
+        iframe.id = 'receiver';
+        iframe.src = 'https://pub.direct/sync.html';
+        document.body.appendChild(iframe);
+        iframe.onload = ifLoaded;
+      }
+
+      function IDGenerator() {
+        var array = new Uint32Array(8);
+        window.crypto.getRandomValues(array);
+        var str = '';
+        for (var i = 0; i < array.length; i++) {
+          str += (i < 2 || i > 5 ? '' : '-') + array[i].toString(16).slice(-4);
+        }
+        return str;
+      }
+
+      function ifLoaded() {
+        if (consent === true) {
+          if (doNotTrackUser === true) {
+            messageHub('doNotTrackUser', parseData);
+          }
+          // Get the window displayed in the iframe.
+          if (iframeCreated === false) {
+            messageHub('self', parseData);
+          } else if (iframeCreated === true) {
+            messageHub(incomingUserId, parseData);
+          }
+        } else {
+
+        }
+      }
+
+      function msgFiltering(e) {
+        iframeCreated = true;
+        incomingUserId = e;
+        if (e.origin !== 'https://pub.direct') {
+          return;
+        }
+        if (iframeCreated === true) {
+          var iframe = document.createElement('iframe');
+          iframe.style.display = 'none';
+          iframe.id = 'receiver';
+          iframe.src = 'https://pub.direct/sync.html';
+          document.body.appendChild(iframe);
+          iframe.onload = ifLoaded;
+        }
+      }
+      function messageHub(e, callback) {
+        if (e !== undefined && e.data !== 'clear' && e !== 'self' && e !== 'doNotTrackUser' && e.data.timestamp !== undefined) {
```

Comment:
This condition is too complex. You can rich the same result by putting this case at the end of function.